src/CreateEntityReducer.js

```diff
@@ -58,6 +58,15 @@ export function createEntityReducer(config) {
 
         var [, actionTypePrefix] = resultKey.toString().match(/(.*)_(FETCH|ERROR|RECEIVE)$/) || [];
 
+
+        if(type === 'ENTITY_DELETE') {
+            if(Iterable.isKeyed(state.getIn(payload))) {
+                return state.setIn(payload.concat('_deleted'), true);
+            } else {
+                return state.deleteIn(payload);
+            }
+        }
+
```

Comment:
I may not be understanding the problem but it seems pretty arbitrary to me that deleting has such different results depending on the data structure. Could we store a list of deleted keys rather than deleting/adding the deleted property?